src/signTxPoolRegistration.c

```
@@ -0,0 +1,836 @@
+{
+	TRACE("UI step %d", subctx->ui_step);
+	ui_callback_fn_t* this_fn = handleOwner_ui_runStep;
+
+	UI_STEP_BEGIN(subctx->ui_step);
+
+	UI_STEP(HANDLE_OWNER_STEP_DISPLAY) {
+		ui_displayOwnerScreen(&subctx->owner, commonTxData->networkId, this_fn);
+	}
+	UI_STEP(HANDLE_OWNER_STEP_RESPOND) {
+		respondSuccessEmptyMsg();
+
+		subctx->currentOwner++;
+		if (subctx->currentOwner == subctx->numOwners) {
+			VALIDATE(subctx->numOwnersGivenByPath == 1, ERR_INVALID_DATA);
+
+			advanceState();
+		}
+	}
+	UI_STEP_END(HANDLE_OWNER_STEP_INVALID);
+}
+
+static void signTxPoolRegistration_handleOwnerAPDU(uint8_t* wireDataBuffer, size_t wireDataSize)
+{
+	{
+		// sanity checks
+		CHECK_STATE(STAKE_POOL_REGISTRATION_OWNERS);
+
+		ASSERT(wireDataSize < BUFFER_SIZE_PARANOIA);
+	}
+
+	explicit_bzero(&subctx->owner, SIZEOF(subctx->owner));
+
+	{
+		// parse data
+		TRACE_BUFFER(wireDataBuffer, wireDataSize);
+
+		read_view_t view = make_read_view(wireDataBuffer, wireDataBuffer + wireDataSize);
+
+		VALIDATE(view_remainingSize(&view) >= 1, ERR_INVALID_DATA);
+		subctx->owner.ownerType = parse_u1be(&view);
+		switch (subctx->owner.ownerType) {
+
+		case SIGN_TX_POOL_OWNER_TYPE_KEY_HASH:
+			VALIDATE(view_remainingSize(&view) == ADDRESS_KEY_HASH_LENGTH, ERR_INVALID_DATA);
+			ASSERT(SIZEOF(subctx->owner.keyHash) == ADDRESS_KEY_HASH_LENGTH);
+			os_memmove(subctx->owner.keyHash, VIEW_REMAINING_TO_TUPLE_BUF_SIZE(&view));
+			TRACE_BUFFER(subctx->owner.keyHash, SIZEOF(subctx->owner.keyHash));
+			break;
+
+		case SIGN_TX_POOL_OWNER_TYPE_PATH:
+			view_skipBytes(&view, bip44_parseFromWire(&subctx->owner.path, VIEW_REMAINING_TO_TUPLE_BUF_SIZE(&view)));
+			// further validation of the path in security policy below
+			TRACE("Owner given by path:");
+			BIP44_PRINTF(&subctx->owner.path);
+			VALIDATE(view_remainingSize(&view) == 0, ERR_INVALID_DATA);
+
+			subctx->numOwnersGivenByPath++;
+			VALIDATE(subctx->numOwnersGivenByPath <= 1, ERR_INVALID_DATA);
+
+			break;
```

I know for an unsuported "owner type" the UI logic would fail, but for the sake of clarity/"foolproofness", I would expect this switch to throw if an unsupported pool owner is passed